src/mscorlib/src/System/Globalization/CompareInfo.Invariant.cs

```diff
@@ -0,0 +1,238 @@
+                    char firstValueChar = InvariantToUpper(value[0]);
+                    for (ctrSource = lastSourceStart; ctrSource >= 0; ctrSource--)
+                    {
+                        sourceChar = InvariantToUpper(source[ctrSource]);
+                        if (sourceChar != firstValueChar)
+                        {
+                            continue;
+                        }
+                        for (ctrValue = 1; ctrValue < valueCount; ctrValue++)
+                        {
+                            sourceChar = InvariantToUpper(source[ctrSource + ctrValue]);
+                            valueChar = InvariantToUpper(value[ctrValue]);
+
+                            if (sourceChar != valueChar)
+                            {
+                                break;
+                            }
+                        }
+
+                        if (ctrValue == valueCount)
+                        {
+                            return ctrSource;
+                        }
+                    }
+                }
+                else
+                {
+                    char firstValueChar = value[0];
+                    for (ctrSource = lastSourceStart; ctrSource >= 0; ctrSource--)
+                    {
+                        sourceChar = source[ctrSource];
+                        if (sourceChar != firstValueChar)
+                        {
+                            continue;
+                        }
+
+                        for (ctrValue = 1; ctrValue < valueCount; ctrValue++)
+                        {
+                            sourceChar = source[ctrSource + ctrValue];
+                            valueChar = value[ctrValue];
+
+                            if (sourceChar != valueChar)
+                            {
+                                break;
+                            }
+                        }
+
+                        if (ctrValue == valueCount)
+                        {
+                            return ctrSource;
+                        }
+                    }
+                }
+            }
+
+            return -1;
+        }
+
+        private static char InvariantToUpper(char c)
+        {
+            return ('a' <= c && c <= 'z') ? (char)(c - 0x20) : c;
+        }
+
+        private unsafe SortKey InvariantCreateSortKey(string source, CompareOptions options)
+        {
+            if (source == null) { throw new ArgumentNullException(nameof(source)); }
+            Contract.EndContractBlock();
+
+            if ((options & ValidSortkeyCtorMaskOffFlags) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
+            }
+
+            byte [] keyData;
+            if (source.Length == 0)
+            { 
+                keyData = EmptyArray<Byte>.Value;
+            }
+            else
+            {
+                // As we word as ordinal, then generate the sort key as the codepoint ordinal values
```

Comment:
in the invariant mode, all string comparison are done as ordinal. so when generating the sort keys we generate it according to this fact. so comparing the sort keys will be same as comparing teh actual strings. I can change the comment to be clearer. I agree the comment is not clear as I meant to say "compare" instead of "word" :-(